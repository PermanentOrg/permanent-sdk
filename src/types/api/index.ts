export * from './AccountVo';
export * from './ArchiveVo';
export * from './BaseVo';
export * from './FileVo';
export * from './FolderVo';
export * from './PresignedPost';
export * from './RecordVo';
export * from './S3UploadVo';
