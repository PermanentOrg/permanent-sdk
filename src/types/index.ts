export * from './api';
export * from './sdk';
export * from './stela';
