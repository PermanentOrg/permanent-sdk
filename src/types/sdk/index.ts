export * from "./AccessRole";
export * from "./Account";
export * from "./Archive";
export * from "./ArchiveRecord";
export * from "./ArchiveRecordType";
export * from "./ClientConfiguration";
export * from "./DerivativeType";
export * from "./File";
export * from "./FileSystemItem";
export * from "./Folder";
export * from "./ShareLink";
export * from "./Status";
