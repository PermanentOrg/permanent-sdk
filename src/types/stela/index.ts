export * from "./Items";
export * from "./PaginatedItems";
export * from "./StelaFile";
export * from "./StelaFolder";
export * from "./StelaRecord";
