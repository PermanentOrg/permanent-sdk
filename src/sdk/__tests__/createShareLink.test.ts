import nock from 'nock';
import { createShareLink } from '..';
import { AccessRole } from '../../types';

describe('createShareLink', () => {
  test('should create a share link', async () => {
    nock('https://permanent.local')
      .post(
        '/api/share/generateShareLink',
        {
          folderLinkId: 8306,
          maxUses: 0,
          previewToggle: false,
          defaultAccessRole: 'access.role.viewer',
        },
      )
      .replyWithFile(
        200,
        `${__dirname}/fixtures/createShareLink/shareLinkVo.json`,
        { 'Content-Type': 'application/json' },
      );

    const shareLink = await createShareLink(
      {
        bearerToken: '12345',
        baseUrl: 'https://permanent.local/api',
      },
      { fileSystemId: 8306 },
      0,
      false,
      AccessRole.Viewer,
    );

    expect(shareLink).toMatchSnapshot();
  });
});
