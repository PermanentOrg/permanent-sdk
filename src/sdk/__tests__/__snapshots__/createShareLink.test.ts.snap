// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createShareLink should create a share link 1`] = `
{
  "autoApprove": false,
  "byAccountId": 1770,
  "createdAt": 2024-01-31T23:41:54.000Z,
  "defaultAccessRole": "access.role.viewer",
  "expiresAt": undefined,
  "id": 173,
  "maxUses": 0,
  "sharedFileSystemItemId": 8306,
  "showPreview": false,
  "status": "status.generic.ok",
  "updatedAt": 2024-01-31T23:41:54.000Z,
  "url": "https://local.permanent.org/share/b6b457c4c869d05bb8977ea4be6e66d3957442ce7dcc5892354faf084557a805",
  "urlToken": "b6b457c4c869d05bb8977ea4be6e66d3957442ce7dcc5892354faf084557a805",
  "uses": 0,
}
`;
